src/Functions/lemmatize.cpp

```diff
@@ -0,0 +1,118 @@
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypeString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <Functions/Lemmatizers.cpp>
+#include <Interpreters/Context.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+namespace
+{
+
+struct LemmatizeImpl
+{
+    static void vector(
+        const ColumnString::Chars & data,
+        const ColumnString::Offsets & offsets,
+        ColumnString::Chars & res_data,
+        ColumnString::Offsets & res_offsets,
+        Lemmatizers::LemmPtr & lemmatizer)
+    {
+        res_data.resize(data.size());
+        res_offsets.assign(offsets);
+
+        UInt64 data_size = 0;
+        for (UInt64 i = 0; i < offsets.size(); ++i)
+        {
+            /// lemmatize() is based on fact that string ends with '\0' 
+            auto result = lemmatizer->lemmatize(reinterpret_cast<const char *>(data.data() + offsets[i - 1]));
+            size_t new_size = strlen(result.get()) + 1;
+            
+            if (data_size + new_size > res_data.size())
+                res_data.resize(data_size + new_size);
+
+            memcpy(res_data.data() + data_size, reinterpret_cast<const unsigned char *>(result.get()), new_size);
+            
+            data_size += new_size;
+            res_offsets[i] = data_size;
+        }
+        res_data.resize(data_size);
+    }
+};
+
+
+class FunctionLemmatize : public IFunction
+{
+public:
+    static constexpr auto name = "lemmatize";
+    static FunctionPtr create(ContextPtr context) {
+        return std::make_shared<FunctionLemmatize>(context->getLemmatizers());
+    }
+
+private:
+    Lemmatizers & lemmatizers;
```

Comment:
Why not `const`?

Reply:
`lemmatizers.getLemmatizer(language)` loads certain dictionary if it wasn't loaded before, therefore function and reference are not const.